src/main/java/seedu/address/logic/command/edit/EditTravelPlanCommand.java

```diff
@@ -85,15 +87,27 @@ public CommandResult execute(Model model) throws CommandException {
      * @param editTravelPlanDescriptor contains updated fields
      * @return TravelPlan to be updated in the travelplan list
      */
-    private static TravelPlan createEditedTravelPlan(TravelPlan travelPlanToEdit,
-                                                     EditDescriptor editTravelPlanDescriptor) {
+    private static TravelPlan createEditedTravelPlan(
+            TravelPlan travelPlanToEdit, EditDescriptor editTravelPlanDescriptor) throws CommandException {
         assert travelPlanToEdit != null;
 
         Name updatedName = editTravelPlanDescriptor.getName().orElse(travelPlanToEdit.getName());
         WanderlustDate updatedStartDate = editTravelPlanDescriptor.getStartDate()
                 .orElse(travelPlanToEdit.getStartDate());
         WanderlustDate updatedEndDate = editTravelPlanDescriptor.getEndDate().orElse(travelPlanToEdit.getEndDate());
 
+        boolean isValidDate = TravelPlan.isValidStartAndEndDate(updatedStartDate, updatedEndDate);
+
+        if (!isValidDate) {
+            throw new CommandException(MESSAGE_INVALID_STARTANDENDDATE);
+        }
+
+        boolean isValidStartDate = TravelPlan.isStartDateAfterToday(updatedStartDate);
```

Comment:
Can this be put in isValidStartAndEndDate()? So, everything is checked in one method?

Reply:
No, different error message